redux/features/prodInfo-slice.js

```diff
@@ -16,10 +22,23 @@ export const prodInfoSlice = createSlice({
       state.visibleImage = payload;
     },
     changeCurrentImage: (state, { payload }) => {
-      state.currentImage = payload;
+      if (typeof payload.isProd !== 'boolean') {
+        state.currentImage.url = payload.url;
+      } else if (!payload.url) {
+        state.currentImage.isProd = payload.isProd;
+      } else if (typeof payload.isProd === 'boolean' && payload.url) {
+        state.currentImage = payload;
+      }
+    },
+    updImageBtnArr: (state, { payload }) => {
+      if (typeof payload.index === 'number' ) {
+        state.imageBtnArr[payload.index] = payload.url;
+      } else {
+        payload.imageBtnArr = payload;
+      }
```

Comment:
updImageBtnArr accepts an optional obj, if you provide an index key you will also need to provide a url key to set the new image btn image.
The buttons on the left of the displayImage